### PR TITLE
RDRP-1038_pnp_run

### PR DESCRIPTION
**Pull request by Ryan2Y79**
## Pull Request submission

*Insert detailed bullet points about your changes here!*

*Insert any instructions to help the reviewer, e.g. "install new requirements from `requirements.txt`"*

*Let the reviewer know what data files are needed (and if applicable, where they are to be found)

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `requirements.txt`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/staging/staging_main.py [hunk 1]
```diff
@@ -249,37 +248,26 @@ def run_staging(  # noqa: C901
...
+            full_responses = helpers.filter_pnp_data(full_responses, config)
 
         if config["global"]["output_full_responses"]:
-            StagingMainLogger.info("Starting output of staged BERD data...")
+            survey_type = config["survey"]["survey_type"]
+            StagingMainLogger.info(f"Starting output of staged {survey_type} data...")
             staging_folder = staging_dict["staging_output_path"]
             tdate = datetime.now().strftime("%y-%m-%d")
             survey_year = config["survey"]["survey_year"]
             staged_filename = (
```
Comment (AnneONS) [inline src/staging/staging_main.py]: I suppose you can use your new function for this now

**Review by AnneONS — COMMENTED** [review]
looks good so far

File: src/staging/staging_main.py (hunk 1 above)
Comment (Ryan2Y79) [reply src/staging/staging_main.py]: You are correct!

Comment (Ryan2Y79) [reply src/staging/staging_main.py]: I tried to preserve the original name format as much as possible.

**Comment by Ryan2Y79** [comment]
I have test run the following user_config conditions to trigger expected responses:

1. survey_type = PBP & load_ni_data = True: **RESPONCE:** EXPECTED RESPONCE where error is flagged that PNP and NI are mutually exclusive.
2. survey_type = PBP & load_ni_data = False: **RESPONCE:** EXPECTED RESPONCE where pipeline ends after mapping.
3. survey_type = BERD & load_ni_data = True: **RESPONCE:** EXPECTED RESPONCE where pipeline ran through fully
4. survey_type = BERD & load_ni_data = False: **RESPONCE:** EXPECTED RESPONCE where pipeline ran through fully

File: src/user_config.yaml [hunk 2]
```diff
@@ -2,18 +2,20 @@ config_validation:
...
+surveys:
+  survey_type: "PNP"
 global:
   # Staging and validation settings
   postcode_csv_check: True
-  load_ni_data: True
+  load_ni_data: False
   # Pipeline run type
-  run_with_snapshot: True                         # Run until point of freezing
+  run_with_snapshot: False                         # Run until point of freezing
```
Comment (AnneONS) [inline src/user_config.yaml]: this should be true

**Review by AnneONS — COMMENTED** [review]
Can you put the top "freezing" setting to "True"?

**Review by AnneONS — COMMENTED** [review]
Hi @Ryan2Y79 please can you check my latest commit and see if you're happy with it? If so then this is ready to merge.

**Comment by Ryan2Y79** [comment]
@AnneONS Those changes look good to me. Thanks for making them. The lack of `f``'s was an oversight on my part. Cheers.

**Review by AnneONS — APPROVED** [review]
all ready to go!

**Pull request merged**